### PR TITLE
SOMA-Agent

### PR DESCRIPTION
**Pull request by mrnolte**
Introduced the pattern to model preferences.
#286 should be merged before this.

### CODE REVIEW

**Comment by mpomarlan** [comment]
Oof how are the diffs always this big? And a lot of the differences are trivial but distracting (blablawholeprefix becomes soma and such).

(And might be nice to have SOMA-Agent in functional notation).

**Comment by mrnolte** [comment]
The differences should be minimal after the merge of #286, as this branch is based on that.

**Comment by mrnolte** [comment]
I now saved SOMA-AGENT as functional Syntax.

File: owl/SOMA-OBJ.owl [hunk 1]
```diff
@@ -1906,9 +1939,19 @@ SubClassOf(SOMA:PathRole SOMA:SpatioTemporalRole)
 AnnotationAssertion(rdfs:comment SOMA:Patient "A role classifying an object that undergoes/is the primary object affected by the event.")
 SubClassOf(SOMA:Patient SOMA:EventAdjacentRole)
 
+
+# Class: SOMA:PerformerRole (Performer role)
+
+AnnotationAssertion(rdfs:comment SOMA:PerformerRole "A role classifying an Agent responsible for performing an Action.
```
Comment (mpomarlan) [inline owl/SOMA-OBJ.owl:1945]: It seems this annotation needs an update to reflect the renaming AgentRole -> PerformerRole.

File: owl/SOMA.owl [hunk 2]
```diff
@@ -489,6 +560,13 @@ SubObjectPropertyOf(SOMA:isSituationIncludedIn <http://www.ontologydesignpattern
 ObjectPropertyDomain(SOMA:isSituationIncludedIn <http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#Situation>)
 ObjectPropertyRange(SOMA:isSituationIncludedIn <http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#Situation>)
 
+# Object Property: SOMA:isSourceFor (is source for)
+
+AnnotationAssertion(rdfs:comment SOMA:isSourceFor "The (transitive) relation between an information object and another which was derived from the former.")
+AnnotationAssertion(rdfs:label SOMA:isSourceFor "is source for")
+SubObjectPropertyOf(SOMA:isSourceFor <http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#associatedWith>)
```
Comment (mpomarlan) [inline owl/SOMA.owl:567]: Should this have domain and range set to InformationObject?

**Review by mpomarlan — CHANGES_REQUESTED** [review]
See comments.

I notice there is a ton of stuff in SOMA.owl. It should not be there, but I think we talked about this previously, and it will be repositioned in some future PR.

File: owl/SOMA-ACT.owl [hunk 3]
```diff
@@ -225,6 +249,8 @@ ObjectPropertyRange(SOMA:hasAction <http://www.ontologydesignpatterns.org/ont/du
 AnnotationAssertion(rdfs:comment SOMA:hasAnswer "The relation between a message and its answer.")
 AnnotationAssertion(rdfs:label SOMA:hasAnswer "has answer")
 SubObjectPropertyOf(SOMA:hasAnswer SOMA:relatesToAnotherRole)
+ObjectPropertyDomain(SOMA:hasAnswer SOMA:Answer)
```
[the remark this answers is not in the document]
Comment (ayden175) [reply owl/SOMA-ACT.owl:252]: Maybe Agent would make more sense for the domain, unless there is another more fitting class?

File: owl/SOMA-ACT.owl [hunk 4]
```diff
@@ -660,7 +826,7 @@ SubClassOf(SOMA:IllocutionaryTask ObjectIntersectionOf(SOMA:CommunicationTask Ob
 # Class: SOMA:Imagining (SOMA:Imagining)
 
 AnnotationAssertion(rdfs:comment SOMA:Imagining "A Mental task in which the Agent constructs a mental representation of a possible world. An Agent performing an Imagining activity does not aim to construct a representation that aspires to be faithful to some past, present, or future state of affairs of the actual world it is embodied in.")
-SubClassOf(SOMA:Imagining SOMA:MentalTask)
+SubClassOf(SOMA:Imagining SOMA:InformationAcquisition)
```
[the remark this answers is not in the document]
Comment (mrnolte) [reply owl/SOMA-ACT.owl:829]: Going by the Definition of InformationAcquisition, which is that an Agent acquires new Information, i.e., there is an output role Knowledge, and that Imagining constructs a mental representation of some (possibly future) state, which then would be the output, I sorted it here. Where would you put it instead?

File: owl/SOMA-ACT.owl (hunk 3 above)
[the remark this answers is not in the document]
Comment (mrnolte) [reply owl/SOMA-ACT.owl:252]: Was an error, fixed it. Should be Message.

File: owl/SOMA-ACT.owl [hunk 5]
```diff
@@ -163,54 +196,45 @@ Declaration(NamedIndividual(SOMA:ExecutionState_Succeeded))
...
-This relation is seen as transitive."@en)
-AnnotationAssertion(rdfs:label :causes "causes"@en)
-SubObjectPropertyOf(:causes <http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#precedes>)
-TransitiveObjectProperty(:causes)
+AnnotationAssertion(rdfs:label SOMA:answers "The relation between an answering message and the message it answers.")
+AnnotationAssertion(rdfs:label SOMA:answers "answers")
+SubObjectPropertyOf(SOMA:answers SOMA:relatesToAnotherRole)
+InverseObjectProperties(SOMA:answers SOMA:hasAnswer)
+ObjectPropertyDomain(SOMA:answers SOMA:Message)
+ObjectPropertyRange(SOMA:answers SOMA:Message)
```
[the remark this answers is not in the document]
Comment (mrnolte) [reply owl/SOMA-ACT.owl:206]: This was an error, answers has as domain answer and as range message, and the inverse is true for hasAnswer. Fixed it.

File: owl/SOMA-ACT.owl [hunk 6]
```diff
@@ -1132,4 +1445,9 @@ SubClassOf(Annotation(rdfs:comment "If something is an initial situation of some
...
 SubClassOf(Annotation(rdfs:comment "If a transition has a terminal situation then it also has a manifestation.") ObjectSomeValuesFrom(SOMA:hasTerminalSituation <http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#Situation>) ObjectSomeValuesFrom(<http://www.ease-crc.org/ont/SOMA.owl/manifestsIn> <http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#Event>))
+SubObjectPropertyOf(ObjectPropertyChain(SOMA:causesTransition SOMA:hasInitialSituation <http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#includesEvent>) SOMA:terminates)
+SubObjectPropertyOf(ObjectPropertyChain(SOMA:causesTransition SOMA:hasTerminalScene <http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#includesEvent>) SOMA:directlyCauses)
+SubObjectPropertyOf(ObjectPropertyChain(<http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#includesEvent> SOMA:hasInitialSituation SOMA:hasTerminalSituation <http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#includesEvent>) SOMA:isReplacedBy)
```
[the remark this answers is not in the document]
Comment (mrnolte) [reply owl/SOMA-ACT.owl:1450]: Fixed it

**Pull request merged**